sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -104,6 +104,8 @@ abstract class ExternalCatalog {
    */
   def alterTable(tableDefinition: CatalogTable): Unit
 
+  def alterTableSchema(db: String, table: String, schema: StructType): Unit
```

Comment:
Wasn't sure if I needed to add a doc comment here since this appears to be missing for most methods in this interface but I'll add one here since there are preconditions the caller must be aware of